promql/bench_test.go

```diff
@@ -265,6 +279,17 @@ func BenchmarkRangeQuery(b *testing.B) {
 				if res.Err != nil {
 					b.Fatal(res.Err)
 				}
+				m, err := res.Matrix()
+				if err != nil {
+					b.Fatal(res.Err)
+				}
+				inMemoryBytes := 0
+				for _, s := range m {
+					inMemoryBytes += cap(s.Floats) * fPointSize
+					inMemoryBytes += cap(s.Histograms) * hPointSize
+				}
+
+				b.ReportMetric(float64(inMemoryBytes), "QueryInMemoryBytes")
```

Comment:
It's some subset of all the query memory, right?  `"PointSlicesBytes"` ?

Reply:
Yeah... I was thinking in keep track the peak of sum of the slice capacity used in a query (keeping track on the putFPointSlice and getFPointSlice functions). But in order to do so i would need to make those functions part of the `evaluator` and adding those metrics on the `samplesStats`... 

Im not sure if this is worth it so i did this way as is way less intrusive.

Reply:
I just noticed it requires `unsafe`.  Now I feel it would be better to take it out, since you proved the point.